test/defaultToAny.test.js

```diff
@@ -0,0 +1,92 @@
+  describe('defaults to value', () => {
+    it('checked are null, NaN', () => {
+      const result = defaultToNumber([null, NaN]);
+      expect(result).toBe(defaultNumber);
+    });
+    it('checked are null, undefined', () => {
+      const result = defaultToNumber([null, undefined]);
+      expect(result).toBe(defaultNumber);
+    });
+    it('checked are undefined, NaN', () => {
+      const result = defaultToNumber([undefined, NaN]);
+      expect(result).toBe(defaultNumber);
+    });
+    it('checked are undefined, null, NaN', () => {
+      const result = defaultToNumber([undefined, null, NaN]);
+      expect(result).toBe(defaultNumber);
+    });
+
+    it('returns the first value that is not NaN, null or undefined', () => {
+      const result = defaultToAny(undefined, 1, null, 10);
+      expect(result).toBe(1);
+    });
+  });
+
+  describe('does not default to value', () => {
+    it('checked is a number', () => {
+      const result = defaultToAny(0, -1, defaultNumber);
+      expect(result).toBe(0);
+    });
+    it('checked is a boolean', () => {
+      const result = defaultToAny(false, true, defaultNumber);
+      expect(result).toBe(false);
+    });
+    it('checked is a date', () => {
+      const result = defaultToAny(new Date(), 'hello', defaultNumber);
+      expect(result).toBeDate();
+    });
+    it('checked is a function', () => {
+      const result = defaultToAny(() => {}, 'hello', defaultNumber);
+      expect(result).toBeFunction();
+    });
+    it('checked is a string', () => {
+      const result = defaultToAny('hello', 'world', defaultNumber);
+      expect(result).toBe('hello');
+    });
+    it('checked is an empty string', () => {
+      const result = defaultToAny('', 'hello');
+      expect(result).toBe('');
+    });
+    it('checked is an empty object', () => {
+      const result = defaultToAny({}, 'hello', defaultNumber);
+      expect(result).toBeEmptyObject();
+    });
+    it('checked is an empty array', () => {
+      const result = defaultToAny([], 'hello', defaultNumber);
+      expect(result).toBeArrayOfSize(0);
+    });
+
+    it('returns the first value that is not NaN, null or undefined', () => {
+      const result = defaultToAny('hello', NaN, null, undefined);
+      expect(result).toBe('hello');
+    });
+  });
+
+  describe('can handle one parameter', () => {
+    it('value is not NaN, null or undefined', () => {
+      const result = defaultToAny(1);
+      expect(result).toBe(1);
+    });
+    /* It is somewhat unclear what should be returned in
+     * cases where the default value is an empty array. */
+    it('value is NaN', () => {
+      const result = defaultToAny(NaN);
+      expect(result).toBeArrayOfSize(0);
+    });
+    it('value is null', () => {
+      const result = defaultToAny(null);
+      expect(result).toBeArrayOfSize(0);
+    });
+    it('value is undefined', () => {
+      const result = defaultToAny(undefined);
+      expect(result).toBeArrayOfSize(0);
+    });
```

Comment:
These return themselves, which is a bit unexpected. This is a corner case though and I'm not sure what the correct return value should be. I went with an empty array, because that's what defaultValues are in `defaultToAny.js` in this case. Thoughts?

Reply:
You're not giving any default values, which would suggest this is undefined behaviour in which case it should throw an IllegalArgumentException or similar